### ISSUE TITLE
Connection to mesos framework stops working after a while

### ISSUE THREAD

**Issue by chr0n1x** [jenkinsci/mesos-plugin#234]
I have a mesos 0.27 cluster, jenkins 2.0, mesos plugin version 0.12.

Having a configuration that keeps a persistent connection to mesos seems to just fail eventually (i.e.: on-demand framework registration is off) and jenkins fails to spin up any tasks in mesos. What's strange is that jenkins still shows up in mesos as an active framework.

All I'm seeing in the mesos logs is 

```
W0425 21:55:46.937744  1351 master.cpp:5548] Failed to authenticate scheduler-8e1d560c-62d3-4102-b77f-e1410651cf00@10.6.202.16:59579: Authentication discarded
```

The strange thing though is that the credentials that are being used to authenticate **did** work upon initial connection. Changing the configuration & forcing the driver to reload does not seem to help either. This includes changing the configuration from non-on-demand framework registration to on-demand.

Not sure if this is worth mentioning, but the framework password has special characters in it (`$`)

Not quite sure where to even start looking to debug, any help would be appreciated!

**Comment by jeschkies** [comment]
@chr0n1x, do you see the issues with 1.0 or 2.0 of the plugin? Would you mind giving it a try?

**Comment by chr0n1x** [comment]
@jeschkies hey, this was specifically with version 0.12. My team has migrated off of Jenkins now though so I don't need to test. I bet it's fixed in 1.0 or 2.0 so I'll close 😄
